java-maven/maven-agent/pom.xml

```diff
@@ -64,6 +64,28 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>pl.project13.maven</groupId>
+                <artifactId>git-commit-id-plugin</artifactId>
+                <version>4.0.0</version>
+                <executions>
+                    <execution>
+                        <id>get-the-git-infos</id>
+                        <goals>
+                            <goal>revision</goal>
+                        </goals>
+                        <phase>initialize</phase>
+                    </execution>
+                </executions>
+                <configuration>
+                    <generateGitPropertiesFile>true</generateGitPropertiesFile>
+                    <generateGitPropertiesFilename>${project.build.outputDirectory}/git.properties</generateGitPropertiesFilename>
```

Comment:
@LiranLast this will create `git.properties` file in the .jar 
and the content will be like this:
```
#Generated by Git-Commit-Id-Plugin
#Fri Apr 24 17:25:41 IDT 2020
git.commit.id.full=453500ed7e46ed210400f018a58dcf59ce085788
```

Reply:
Check out the build number maven plugin, it seems to have cleaner and nicer ways to embed into the manifest. You can see some links in #SDK

Reply:
@LiranLast the buildnumber-maven-plugin does look cleaner and more of a standard plugin. But it doesn't get the remote origin, so if we use buildnumber-maven-plugin then the user will have to enter the remote origin url into the pom.xml (like we ask them to do in the perforce integration).

Reply:
@ElDuderinos it doesn't sound all that bad.
Using this plugin will require SDK updates, which sounds like more work at this point

Reply:
Its not a lot of work in the SDK and less work for the costumers .
@LiranLast @ElDuderinos